e2e/test/iothub/SasCredentialAuthenticationTests.cs

```diff
@@ -0,0 +1,225 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.Common.Exceptions;
+using Microsoft.Azure.Devices.E2ETests.Helpers;
+using Microsoft.Azure.Devices.Shared;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+#if !NET451
+
+using Microsoft.Rest;
+using Azure;
+
+#endif
+
+using ClientOptions = Microsoft.Azure.Devices.Client.ClientOptions;
+
+namespace Microsoft.Azure.Devices.E2ETests.iothub
+{
+    /// <summary>
+    /// Tests to ensure authentication using SAS credential succeeds in all the clients.
+    /// </summary>
+    [TestClass]
+    [TestCategory("E2E")]
+    [TestCategory("IoTHub")]
+    public class SasCredentialAuthenticationTests : E2EMsTestBase
+    {
+        private readonly string _devicePrefix = $"E2E_{nameof(SasCredentialAuthenticationTests)}_";
+
+#if !NET451
+
+        [Ignore]
+        [LoggedTestMethod]
+        public async Task RegistryManager_Http_SasCredentialAuth_Success()
+        {
+            // arrange
+            string signature = Configuration.IoTHub.GetIotHubSharedAccessSignature(TimeSpan.FromHours(1));
+            using var registryManager = RegistryManager.Create(
+                Configuration.IoTHub.GetIotHubHostName(),
+                new AzureSasCredential(signature));
+
+            var device = new Device(Guid.NewGuid().ToString());
+
+            // act
+            Device createdDevice = await registryManager.AddDeviceAsync(device).ConfigureAwait(false);
+
+            // assert
+            Assert.IsNotNull(createdDevice);
+
+            // cleanup
+            await registryManager.RemoveDeviceAsync(device.Id).ConfigureAwait(false);
+        }
+
+        [Ignore]
+        [LoggedTestMethod]
+        public async Task RegistryManager_Http_SasCredentialAuth_Renewed_Success()
+        {
+            // arrange
+            string signature = Configuration.IoTHub.GetIotHubSharedAccessSignature(TimeSpan.FromHours(-1));
+            var sasCredential = new AzureSasCredential(signature);
+            using var registryManager = RegistryManager.Create(
+                Configuration.IoTHub.GetIotHubHostName(),
+                sasCredential);
+
+            var device = new Device(Guid.NewGuid().ToString());
+
+            // act
+            try
+            {
+                await registryManager.AddDeviceAsync(device).ConfigureAwait(false);
```

Comment:
I'd add a failure case for if this line executes without exception. Maybe throw an AssertionFailedException("Expected previous line to throw an unauthorized exception");

Reply:
yes that makes sense